templates/admin.html

```diff
@@ -257,41 +264,95 @@ <h2 id="demotesttts">Demo/Test TTS</h2>
-            <button type="submit">Generate TTS</button>
+            <button id="submit" type="submit">
+                Generate TTS
+            </button>
         </form>
     </div>
 
     <script>
+        // Audio player
+        const audioPlayer = document.getElementById('audioSource');
+        function enableLoader(enable) {
+            // Change the submit button text and disable it while loading
+            const submit = document.getElementById('submit');
+            submit.disabled = enable ? true : false;
+            submit.style.opacity = enable ? 0.7 : 1.0;
+            submit.innerText = enable ? 'Generating...' : 'Generate TTS';
+            // Disable the audio player while loading
+            audioPlayer.disabled = enable ? true : false;
+            audioPlayer.style.opacity = enable ? 0.7 : 1.0;
+            audioPlayer.style.pointerEvents = enable ? 'none' : 'auto';
+        }
+        audioPlayer.addEventListener('canplay', () => {
+            enableLoader(false);
+            audioPlayer.play();
+        });
+        audioPlayer.addEventListener('abort', () => {
+            enableLoader(false);
+        });
+        audioPlayer.addEventListener('error', (e) => {
+            enableLoader(false);
+        });
+
+        // Streaming selector
+        const streaming = document.getElementById('streaming');
+        streaming.addEventListener('change', (event) => {
+            const outputFileContainer = document.getElementById('outputFileContainer');
+            if (event.target.value === 'true') {
+                outputFileContainer.style.display = 'none';
+            } else {
+                outputFileContainer.style.display = 'flex';
+            }
+        });
+        // Form submit
         const form = document.getElementById('ttsForm');
         form.addEventListener('submit', async (event) => {
             event.preventDefault();
 
+            enableLoader(true);
+            audioPlayer.pause();
+            
             const formData = new FormData(form);
-            const response = await fetch('/tts-demo-request', {
-                method: 'POST',
-                body: formData
-            });
-
-            const result = await response.json();
-
-            // Update the generated audio file path
-            const outputFilePath = document.getElementById('outputFilePath');
-            outputFilePath.textContent = result.output_file_path;
-
-            // Update the audio player source
-            const audioPlayer = document.getElementById('audioSource');
-            audioPlayer.src = `/audio/${result.output_file_path}`;
-            audioPlayer.load(); // Reload the audio player
-            audioPlayer.play(); // Play the audio;
+            if (formData.get('streaming') === 'true') {
+                const searchParams = new URLSearchParams(formData).toString();
+                audioPlayer.src = "/tts-demo-request?" + searchParams;
```

Comment:
@rbruels Hello, I am not sure what I was missing before but I had to change `"tts-demo-request?"` to `"http://127.0.0.1:7851/tts-demo-request?"` for the Generate button to work.

Just wanted to bring this to your attention.

Reply:
possibly need http://{{ params.ip_address }}:{{ params.port_number }} pushing in.

Reply:
Good call!